massa-network-worker/src/handshake_worker.rs

```diff
@@ -25,18 +26,183 @@ use massa_models::{
+                            connection_id,
+                            worker,
+                        }) => {
+                            let result_tx = result_tx.clone();
+                            // Spawn an async task to run the handshake.
+                            let handle = runtime_handle.spawn(async move {
+                                let result = worker.run().await;
+                                // Spawn a blocking task to send the result back on the channel.
+                                Handle::current()
+                                    .spawn_blocking(move || {
+                                        result_tx
+                                            .send(HandshakeResultMsg::HandshakeReturnType{connection_id, result})
+                                            .expect("Failed to send handshake return type message.")
+                                    })
+                                    .await.expect("Failed to spawn blocking call.");
+                            });
+                            task_handles.insert(connection_id, handle);
+                        },
+                        Ok(HandshakeManagerMsg::SendPeerList{peer_list, ip, reader, writer}) => {
+                            if cfg.max_in_connection_overflow > handshake_peer_list_futures.len() {
+                                let result_tx = result_tx.clone();
+                                let msg = Message::PeerList(peer_list);
+                                let timeout = cfg.peer_list_send_timeout.to_duration();
+                                let max_bytes_read = cfg.max_bytes_read;
+                                let max_bytes_write = cfg.max_bytes_write;
+                                let max_ask_blocks = cfg.max_ask_blocks;
+                                let max_operations_per_block = cfg.max_operations_per_block;
+                                let thread_count = cfg.thread_count;
+                                let endorsement_count = cfg.endorsement_count;
+                                let max_advertise_length = cfg.max_peer_advertise_length;
+                                let max_endorsements_per_message = cfg.max_endorsements_per_message;
+                                let max_operations_per_message = cfg.max_operations_per_message;
+                                let max_message_size = cfg.max_message_size;
+                                let max_datastore_value_length = cfg.max_datastore_value_length;
+                                let max_function_name_length = cfg.max_function_name_length;
+                                let max_parameters_size = cfg.max_parameters_size;
+                                let max_op_datastore_entry_count = cfg.max_op_datastore_entry_count;
+                                let max_op_datastore_key_length = cfg.max_op_datastore_key_length;
+                                let max_op_datastore_value_length = cfg.max_op_datastore_value_length;
```

Comment:
Was reading this PR to give me some guidance/reference impl/inspiration for something I'm working on, and thought this comment might be worth sharing while I was here (feel free to resolve without change). 

where it was `let foo = cfg.foo;`, the line is now `foo, // cfg.foo;` (except for the fields with renames/conversions)

timeout is the exception in order to preserve the conversion.

```suggestion
    let NetworkConfig {
        peer_list_send_timeout: timeout, // = cfg.peer_list_send_timeout.to_duration();
        max_bytes_read,                  // = cfg.max_bytes_read;
        max_bytes_write,                 // = cfg.max_bytes_write;
        max_ask_blocks,                  // = cfg.max_ask_blocks;
        max_operations_per_block,        // = cfg.max_operations_per_block;
        thread_count,                    // = cfg.thread_count;
        endorsement_count,               // = cfg.endorsement_count;
        max_peer_advertise_length: max_advertise_length, // = cfg.max_peer_advertise_length;
        max_endorsements_per_message,    // = cfg.max_endorsements_per_message;
        max_operations_per_message,      // = cfg.max_operations_per_message;
        max_message_size,                // = cfg.max_message_size;
        max_datastore_value_length,      // = cfg.max_datastore_value_length;
        max_function_name_length,        // = cfg.max_function_name_length;
        max_parameters_size,             // = cfg.max_parameters_size;
        max_op_datastore_entry_count,    // = cfg.max_op_datastore_entry_count;
        max_op_datastore_key_length,     // = cfg.max_op_datastore_key_length;
        max_op_datastore_value_length,   // = cfg.max_op_datastore_value_length;
    } = cfg;
    let timeout = timeout.to_duration();
```

Reply:
Nice one, thanks! I've incorporated it, with a few changes.